rotkehlchen/chain/evm/decoding/decoder.py

```diff
@@ -880,6 +885,16 @@ def _get_eth_transfer_event(self, tx: EvmTransaction) -> Optional['EvmEvent']:
             counterparty=counterparty,
         )
 
+    def _get_transfer_or_approval_token_kind(self, tx_log: EvmTxReceiptLog) -> EvmTokenKind | None:
+        """Determine if a transfer or approval event is for an erc20 or erc721 token."""
+        if self._is_non_conformant_erc721(tx_log.address) or len(tx_log.topics) == 4:  # typical ERC721 has 3 indexed args  # noqa: E501
+            return EvmTokenKind.ERC721
+        elif len(tx_log.topics) == 3:  # typical ERC20 has 2 indexed args
+            return EvmTokenKind.ERC20
```

Comment:
is there a reason to not put the erc20 before? I guess is the most common type of token and you save 2 checks for any erc20